### DISCUSSION TITLE
Change Citation/References behaviour

### DISCUSSION THREAD

**Discussion by SebastianBodza** [markdoc/markdoc#515] [General]
I would like to change the default behaviour of citations / references in the style of: 
```
[Some super relevant information][1] and some more Text. 

# References: 
[1]: https://somelink.com
```

Markdoc simply replaces the `Some super relevant information` with a link and removes the References. 

Is there a way to customize the behaviour? So to change the link styling specific to the references (and not all general links) as well as keeping the section below # References with a specific styling? 

Thanks! 👋

**Comment by matv-stripe** [comment]
Hi @SebastianBodza ! This behavior is defined in the CommonMark spec, which we use in the parser of markdown-it, (the underlying markdown parsing library). Here is a code pointer where I believe this is happening: https://github.com/markdown-it/markdown-it/blob/master/lib/rules_inline/link.mjs#L83C7-L83C21

Does adding backslashes to escape the `[` and `]` fix your issue?
![CleanShot 2024-07-09 at 10 16 40@2x](https://github.com/markdoc/markdoc/assets/55463370/63e20eac-0a17-401a-b171-7fe2d0a55bd6)

(https://markdown-it.github.io/ is the site I am using in the above screenshot)
